cropduster/models.py

```diff
@@ -399,13 +399,13 @@ def get_image_size(self, size_name=None):
                 return (thumb.width, thumb.height)
 
         # Get the original size
-        if not self.image or not default_storage.exists(safe_str_path(self.image.path)):
+        if not self.image or not default_storage.exists(safe_str_path(self.image.name)):
             return (0, 0)
         elif self.width and self.height:
             return (self.width, self.height)
         else:
             try:
-                img = PIL.Image.open(safe_str_path(self.image.path))
+                img = PIL.Image.open(safe_str_path(self.image.name))
```

Comment:
This won't work (you're trying to open a PIL.Image using the image's filename relative to the s3 bucket / media_root). You'll need to use django file storage to open it, similar to your exists change above. `self.image` is an instance of django's `ImageFieldFile`, which means it has automatic hooks into default_storage. For example `f` in `with default_storage.open(self.image.name) as f:` is the same object as `self.image.file`.

So, the better way to write this using django file storage APIs would be:

```python
try:
    img = PIL.Image.open(self.image.file)
except (IOError, ValueError, TypeError):
    return (0, 0)
else:
    return img.size

Reply:
Actually, I just realized that `self.image.file` might not be 100% reliable if something elsewhere in the code somehow closed that file handle. The safest way to turn an ImageFieldFile to a PIL.Image is:

```python
try:
    with self.image as f:
        f.open()
        img = PIL.Image.open(BytesIO(f.read()))
except (IOError, ValueError, TypeError):
    return (0, 0)
else:
    return img.size
```